### ISSUE TITLE
SaveChanges closes connection opened by user using EF API

### ISSUE THREAD

**Issue by JonPSmith** [aspnet/EntityFramework#6968]
I am using Sqlite in-memory for unit testing. I have upgraded from v1.0.1 to 1.1.0-preview1-final and I now get the error `SQLite Error 1: 'no such table:  xxx`.

### Steps to reproduce
1. Run the code under  SqlLite v1.0.1 and SqlServer v1.0.1 it works fine
2. Upgrade to SqlLite 1.1.0-preview1-final and SqlServer 1.1.0-preview1-final and it fails with `SQLite Error 1: 'no such table:  xxx`. on the unit test line:
```
db.BookAndAuthors.Count().ShouldEqual(4);
```
3. Downgrade to v1.0.1 and it works again.

The SqlLite code is 

```csharp
        [Fact]
        public void TestWriteTestDataSqliteOk()
        {
            //SETUP
            var inMemDb = new SqliteInMemory();

            //ATTEMPT
            using (var db = inMemDb.GetContextWithSetup())
            {
                var books = EfTestData.CreateFourBooks();
                db.BookAndAuthors.AddRange(books);
                db.SaveChanges();

                //VERIFY
                db.BookAndAuthors.Count().ShouldEqual(4);
            }
        }
```
The SqliteInMemory class contains

```csharp
public EfCoreContext GetContextWithSetup()
{
    var db = new EfCoreContext(InMemorySqlite());
    db.Database.OpenConnection();
    db.Database.EnsureCreated();

    var serviceProvider = db.GetInfrastructure();
    serviceProvider.GetService(typeof(ILoggerFactory));
    var loggerFactory = (ILoggerFactory)serviceProvider.GetService(typeof(ILoggerFactory));
    loggerFactory.AddProvider(new MyLoggerProvider(_logs));

    return db;
}

//---------------------------------------------------
//private methods

private DbContextOptions<EfCoreContext> InMemorySqlite()
{
    //Thanks to https://www.scottbrady91.com/Entity-Framework/Entity-Framework-Core-In-Memory-Testing
    var connectionStringBuilder =
        new SqliteConnectionStringBuilder { DataSource = ":memory:" };
    ConnectionString = connectionStringBuilder.ToString();
    var connection = new SqliteConnection(ConnectionString);

    // create in-memory db
    var builder = new DbContextOptionsBuilder<EfCoreContext>();
    builder.UseSqlite(connection);

    return builder.Options;
}
```

### The issue
I get an exception `SQLite Error 1: 'no such table:  xxx`. on the unit test line:
```
db.BookAndAuthors.Count().ShouldEqual(4);
```
```
Exception message: SQLite Error 1: 'no such table: BookAndAuthors'.
Stack trace:
at Microsoft.Data.Sqlite.Interop.MarshalEx.ThrowExceptionForRC(Int32 rc, Sqlite3Handle db)
   at Microsoft.Data.Sqlite.SqliteCommand.ExecuteReader(CommandBehavior behavior)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.Execute(IRelationalConnection connection, String executeMethod, IReadOnlyDictionary`2 parameterValues, Boolean closeConnection)
   at Microsoft.EntityFrameworkCore.Storage.Internal.RelationalCommand.ExecuteReader(IRelationalConnection connection, IReadOnlyDictionary`2 parameterValues)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable.Enumerator.BufferlessMoveNext(Boolean buffer)
   at Microsoft.EntityFrameworkCore.Query.QueryMethodProvider.GetResult[TResult](IEnumerable`1 valueBuffers)
   at lambda_method(Closure , QueryContext )
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass19_1`1.<CompileQuery>b__1(QueryContext qc)
   at test.UnitTests.TestEfTestData.TestWriteTestDataSqliteOk() in C:\Users\Jon\Documents\Visual Studio 2015\Projects\EfCoreInAction\test\UnitTests\TestEfTestData.cs:line 45
```

### Further technical details

EF Core version: 1.1.0-preview1-final
Operating system: Windows 10
Visual Studio version: Microsoft Visual Studio Professional 2015
Version 14.0.25425.01 Update 3
Microsoft .NET Framework
Version 4.6.01586

Other details about my project setup: I have a DataLayer containing the classes and application DbContext. The tests are in a top-level `test` directory using XUnit

**Comment by bricelam** [comment]
Possibly due to #6335--we might be looking for the database in a different location.

**Comment by bricelam** [comment]
lol `:memory:` nevermind. 😄

**Comment by ErikEJ** [comment]
In the wrong memory location?? 😆

**Comment by bricelam** [comment]
Does it make a difference if you open the `SqliteConnection` before passing it to `UseSqlite()` (instead of calling `DbContext.Database.OpenConnection()` afterward)?

**Comment by JonPSmith** [comment]
Hi @bricelam,

Yes, opening 

> the SqliteConnection before passing it to UseSqlite() (instead of calling DbContext.Database.OpenConnection() afterward)?

fixed the problem, i.e.
```
var connection = new SqliteConnection(ConnectionString);
connection.Open();
```

... and removing the `db.Database.OpenConnection();` worked a treat. Thanks for that. I will update my code to use that approach.

**Comment by ajcvickers** [comment]
@bricelam Seems like this could be a breaking change?

**Comment by bricelam** [comment]
Yes; re-opening. We need to understand why we're closing the connection in this case now.

**Comment by smitpatel** [comment]
`SaveChanges()` is closing connection in this case.

**Comment by JonPSmith** [comment]
Hi,

I was expecting this bug to be fixed in release 1.1.1 but it doesn't seem to be fixed. Can you clarify:
1. Is it fixed or not in version 1.1.1 of Microsoft.EntityFrameworkCore.Sqlite
2. If its fixed what is the proper way of opening the connection.

Thanks.

**Comment by bricelam** [comment]
This bug is marked as fixed in the 2.0.0 release. You can try it out using [the nightly builds](https://dotnet.myget.org/gallery/aspnetcore-dev)

**Comment by ajcvickers** [comment]
Removing milestone to reconsider for 1.1.1.

**Comment by ajcvickers** [comment]
Discussed in triage and we are going to port this to 1.1.2, so it will ship with the next patch assuming this is approved.

**Comment by smitpatel** [comment]
@ajcvickers - Patch-proposed label?

**Comment by Eilon** [comment]
This patch fix is *approved*. Please follow the normal code review / pull request process and make sure you make the change in the correct branch.

**Comment by ajcvickers** [comment]
Preview builds of this patch fix should be available on the following feeds:

* https://dotnet.myget.org/gallery/aspnet-1-0-5-may2017-patch-public (for fixes in the 1.0.x release)
  * Feed URL to add to NuGet.config: https://dotnet.myget.org/F/aspnet-1-0-5-may2017-patch-public/api/v3/index.json
* https://dotnet.myget.org/gallery/aspnet-1-1-2-may2017-patch-public (for fixes in the 1.1.x release)
  * Feed URL to add to NuGet.config: https://dotnet.myget.org/F/aspnet-1-1-2-may2017-patch-public/api/v3/index.json

If you have a chance, please try out these preview builds and let us know if you have any feedback!

**Comment by AhmedElbatt** [comment]
I have the same problem after upgrading to V 2.0 ..!
Could you please help me for finding a solution ?

**Comment by ajcvickers** [comment]
@AhmedElbatt This issue was fixed before 2.0, so the behavior you are seeing may be due to a different reason. Please file a new issue including a full code listing or project that reproduces what you are seeing.
